checks.d/pgbouncer.py

```diff
@@ -120,38 +120,62 @@ def _collect_stats(self, db, instance_tags):
+                "Please specify a user to connect to PgBouncer as.")
+
+        if host in ('localhost', '127.0.0.1') and password == '':
+            return {  # Use ident method
+                'dsn': "user={} dbname={}".format(user, self.DB_NAME)
+            }
+
+        if port:
+            return {'host': host, 'user': user, 'password': password,
+                    'database': self.DB_NAME, 'port': port}
+
+        return {'host': host, 'user': user, 'password': password,
+                'database': self.DB_NAME}
+
+    def _get_connection(self, key, host='', port='', user='',
+                        password='', database_url='', use_cached=True):
         "Get and memoize connections to instances"
         if key in self.dbs and use_cached:
             return self.dbs[key]
 
-        elif host != "" and user != "":
-            try:
-                if host == 'localhost' and password == '':
-                    # Use ident method
-                    connection = pg.connect("user=%s dbname=%s" % (user, self.DB_NAME))
-                elif port != '':
-                    connection = pg.connect(host=host, port=port, user=user,
-                                            password=password, database=self.DB_NAME)
-                else:
-                    connection = pg.connect(host=host, user=user, password=password,
-                                            database=self.DB_NAME)
-
-                connection.set_isolation_level(pg.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
-                self.log.debug('pgbouncer status: %s' % AgentCheck.OK)
-
-            except Exception:
-                message = u'Cannot establish connection to pgbouncer://%s:%s/%s' % (host, port, self.DB_NAME)
-                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
-                                   tags=self._get_service_checks_tags(host, port),
-                                   message=message)
-                self.log.debug('pgbouncer status: %s' % AgentCheck.CRITICAL)
-                raise
-        else:
-            if not host:
-                raise CheckException("Please specify a PgBouncer host to connect to.")
-            elif not user:
-                raise CheckException("Please specify a user to connect to PgBouncer as.")
+        try:
+            connect_kwargs = self._get_connect_kwargs(
+                host=host, port=port, user=user,
+                password=password, database_url=database_url
+            )
+
+            connection = pg.connect(**connect_kwargs)
+            connection.set_isolation_level(
+                pg.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+            self.log.debug('pgbouncer status: %s' % AgentCheck.OK)
+
+        # re-raise the ChceckExceptions raised by _get_connect_kwargs()
+        except CheckException:
+            raise
+
+        except Exception:
+            message = u'Cannot establish connection to pgbouncer://%s:%s/%s' % (host, port, self.DB_NAME)
```

Comment:
I think if someoneuses the `database_url` to connect this message will be come
`Cannot establish connection to pgbouncer://:/` as it doesn't use the url at all.
Unless i'm missing something ?